.github/workflows/pr.yml

```diff
@@ -15,7 +15,7 @@ jobs:
     name: Secret scan
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@4.2.2
```

Comment:
Need to provide `v`:
```suggestion
      - uses: actions/checkout@v4.2.2
```

And same for others.

Reply:
Do you get an error flagged in VSCode, like this?
<img width="785" alt="Screenshot 2024-12-09 at 12 38 09" src="https://github.com/user-attachments/assets/0d28317a-9f00-496a-ae99-11491cf0d6e6">